### ISSUE TITLE
Inconsistent date format

### ISSUE THREAD

**Issue by KentGrigo** [PeWu/topola-viewer#115]
The format for dates are inconsistent when there is no year.
I've added a screenshot that shows the problem.
The text reads: `8 Sep 1989` and `30 AUG`.
I expected `30 Aug`.

To reproduce, make two related persons, A and B.
Give A the birth date `1989-09-08`.
Give B the birth date `0-08-30`.

![inconsistent-date](https://user-images.githubusercontent.com/5881000/195178198-349a54da-85b5-4eb5-9f58-9f7e1f23bf44.png)

**Comment by PeWu** [comment]
Thanks for the report.
What application are you using to enter the dates?

**Comment by KentGrigo** [comment]
I'm using Gramps.

**Comment by KentGrigo** [comment]
I've added a small test to show the issue.
Just make a new family tree and import the attached .gramps file (GitHub didn't accept that file ending so I added `.txt`, just remove it).
Also, here is a screenshot of the test.

[topola-viewer-115.gramps.txt](https://github.com/PeWu/topola-viewer/files/10065122/topola-viewer-115.gramps.txt)
![topola-viewer-115](https://user-images.githubusercontent.com/5881000/203257426-aa810096-d423-47f6-bd5a-f575514a9b61.png)
